lib/iris/fileformats/pp.py

```diff
@@ -1616,16 +1634,21 @@ def _interpret_fields(fields):
 
         for field in landmask_compressed_fields:
             field.lbrow, field.lbnpt = mask_shape
-            _create_field_data(field, (field.lbrow, field.lbnpt), land_mask)
+            _create_field_data(field, mask_shape,
+                               with_landmask_field=land_mask)
             yield field
 
 
-def _create_field_data(field, data_shape, land_mask):
+def _create_field_data(field, data_shape, with_landmask_field=None):
```

Comment:
@pp-mo Change `with_landmask_field` to `land_mask_field` ... if you agree to the previous comment requests :wink: